pkg/ebpf/events_engine.go

```diff
@@ -0,0 +1,81 @@
+package ebpf
+
+import (
+	"context"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/rules/engine"
+	"github.com/aquasecurity/tracee/types/protocol"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+// engineEvents stage in the pipeline allows rules detection to be executed in the pipeline
+func (t *Tracee) engineEvents(ctx context.Context, in <-chan *trace.Event) (<-chan *trace.Event, <-chan error) {
+	out := make(chan *trace.Event)
+	errc := make(chan error, 1)
+	done := make(chan bool)
+	engineInput := make(chan protocol.Event)
```

Comment:
@NDStrahilevitz That is what I was planning, but wanted more feedback, specially because we want a soft transition or to have an adapter not to break aqua applications using tracee.

Reply:
I'd argue a major version is the time, but we would need to communicate this to other teams and products. Either way, it probably deserves it's own PR, just though it's a good point to raise the issue.

Reply:
> 2\. Remove the `protocol.Event` entirely and just use `trace.Event` directly in signatures.

That would change all signatures in a simple way (and remove the need for checking `event.Payload.(trace.Event)` all the time, so agreed (but for another milestone/PR).